app/models/setting/bootdisk.rb

```diff
@@ -21,12 +21,17 @@ def self.default_settings
         set('bootdisk_generic_efi_host_template', N_('Grub2 template to use for generic EFI host boot disks'),
             'Boot disk Grub2 EFI - generic host', N_('Generic Grub2 EFI image template'), nil, collection: templates),
         set('bootdisk_mkiso_command', N_('Command to generate ISO image, use genisoimage or mkisofs'), 'genisoimage', N_('ISO generation command')),
-        set('bootdisk_cache_media', N_('Installation media files will be cached for full host images'), true, N_('Installation media caching'))
+        set('bootdisk_cache_media', N_('Installation media files will be cached for full host images'), true, N_('Installation media caching')),
+        set('bootdisk_allowed_types', N_('List of allowed bootdisk types, remove type to disable it'), Setting::Bootdisk.bootdisk_types, N_('Allowed bootdisk types'))
       ]
     end
 
     def self.humanized_category
       N_('Boot disk')
     end
+
+    def self.bootdisk_types
+      ['generic', 'host', 'full_host', 'subnet']
+    end
```

Comment:
It's a good practice to use symbols in Ruby, memory management of symbols is way more efficient than strings and they are also immutable.

I suggest to create a helper method here which would do all the logic: `bootdisk_type_allowed?(type_as_symbol)`. This way you can simplify most of the code in this patch.